spec.md

```diff
@@ -1155,18 +1293,36 @@ compilable to non-parameterized Verilog.
 
 ## Passive Types
 
-It is inappropriate for some circuit components to be declared with a type that
-allows for data to flow in both directions. For example, all sub-elements in a
-memory should flow in the same direction. These components are restricted to
-only have a passive type.
+Stateful elements, such as registers and memories, may contain data of aggregate types.
+Registers with bundle types are especially common.
+However, when using bundle types in stateful elements, the notion of `flip` is meaningless.
+There is no directionality to the data inside a register; the data just is.
+
+A passive type is a which does not make use of `flip`.
+More precisely, a passive type is defined recursively:
+
+* All ground types are passive.
+* Probe types are passive.
+* Properties types are passive.
+* A vector type is passive if and only if the element type is passive.
+* A bundle type is passive if and only if it contains no field which is marked `flip` and the type of each field is passive.
+* Enums are passive if and only if each of their variants is passive. (TODO: Is this the proper terminology?)
+
+Registers and memories may only be parametrized over passive types.
+
+
+## Principal Types of Circuit Components
 
-Intuitively, a passive type is a type where all data flows in the same
-direction, and is defined to be a type that recursively contains no fields with
-flipped orientations. Thus all ground types are passive types. Vector types are
-passive if their element type is passive. And bundle types are passive if no
-fields are flipped and if all field types are passive.
+Each circuit component in a module is associated to a type called its principal type.
+This is the type of data that flows through the component.
 
-All property types are passive.
+The principal type is used to determine whether two circuit components may be connected.
+
+For wires, registers, ports, the principal component is declared after the colon `:` when the circuit component is declared.
+
+For nodes, the principal type is inferred from the type of the expression on the left hand side of the equals sign `=` when the node is declared.
+
+For submodule instances, TODO: ???
```

Comment:
is there a type other than Principal Type?

Reply:
@dtzSiFive asked me something similar. I added a sentence to clarify this.

tl;dr -- In something like `wire w : UInt<1>`, the *kind* is `wire` but the *principal type* is `UInt<1>`. It's just there to linguistically emphasize we are speaking formally.

Reply:
Sorry why not then just say "type"?